test/release/examples/primers/ranges.chpl

```diff
@@ -7,245 +7,384 @@
+// bound is specified on the left side of the ``..`` and the high
+// bound is on the right.  These bounds are inclusive for ranges
+// created with ``..``.
+//
+const n = 5,
+      lo = -3,
+      hi = 3;
+writeln("Basic ranges");
+const r = 1..10,    // 1, 2, 3, ..., 10
+      r2 = 0..n,    // 0, 1, 2, ..., n
+      r3 = lo..hi;  // -3, -2, -1, ..., 3
 writeRange(r);
+writeRange(r2);
+writeRange(r3);
+writeln();
 
 // Ranges can also be defined using an `open-interval` form using the
-// ``..<`` operator which causes the upper bound to be omitted.  For
-// example, the range ``3..<10`` is equivalent to ``3..9``.
-writeln("Open interval range rOpen");
-const rOpen = 3..<10;
+// ``..<`` operator, causing the upper bound to be omitted.  For
+// example, the range ``3..<10`` is equivalent to ``3..9``.  This form
+// is particularly useful for 0-based indexing, to avoid bounds like
+// ``n-1``.
+writeln("Open interval ranges");
+const rOpen = 1..<10,  // 1, 2, 3, ..., 9
+      rOpen2 = 0..<n;  // 0, 1, 2, ..., n-1
 writeRange(rOpen);
+writeRange(rOpen2);
+writeln();
+
+
+// Note that whenever a range's low bound exceeds its high bound, the
+// range is considered to be empty.
+writeln("Empty ranges");
+const empty1 = 1..0,
+      empty2 = 0..-1,
+      empty3 = 10..1;
+writeRange(empty1);
+writeRange(empty2);
+writeRange(empty3);
+writeln();
+
+// To create a range representing a decreasing series of values, the
+// ``by`` operator can be used with a negative stride.  As a simple
+// example, the following range represents the integers counting from
+// 10 down to 1.
+
+writeln("Decreasing range");
+const countDown = 1..10 by -1;  // 10, 9, 8, ..., 1
+writeRange(countDown);
+writeln();
 
-// This form can be convenient when iterating over a 0-based
-// data structure using ranges like ``lo..<data.size``.
+// See the `Range Operators` section below for more information about
+// strided ranges and the ``by`` operator.
+
+/*
+   Uses of Ranges
+   --------------
+*/
 
 //
-// Ranges are a basic building block for iteration.  Here we use a
-// for-loop to iterate over the sequence represented by ``r`` and
-// compute the sum of the indices in the sequence.
+// Ranges are a basic building block for iteration.  The following
+// for-loop iterates over the sequence represented by ``r``, computing
+// the sum of the values in the sequence.
 //
 var sum: int = 0;
-for i in r do
+for i in 1..10 do       // compute 1 + 2 + 3 + ... + 10
   sum += i;
-writeln("The sum of the values in ", r, " is ", sum);
-
-//
-// We can compute this same sum with a reduction over the range.
-//
-sum = + reduce r;
-writeln("The sum of the values in ", r, " as computed by reduce is also ", sum);
+writeln("The sum of the values in '1..10' is ", sum);
 writeln();
 
+// Ranges can also drive parallel loop forms such as
+// :ref:`forall<primers-forallLoops>` and
+// :ref:`coforall<primers-taskparallel-coforall>` loops.
```

Comment:
Should we add something about foreach and/or range iteration being order-independent by default?

Reply:
> I could be convinced, though if you want to argue for it (and suggest what I should link to).

I don't feel strongly enough to argue for it. Where my main concern for it are missing features and relatedly, the doc to link for it being in the technotes. Which is too much of an asymmetry as you refer to.